lib/alaveteli_mail_poller.rb

```diff
@@ -0,0 +1,102 @@
+# -*- encoding : utf-8 -*-
+class AlaveteliMailPoller
+  require 'net/pop'
+  include ConfigHelper
+
+  attr_accessor :settings, :failed_mails
+
+  def initialize(values = {})
+    defaults = { :address      => AlaveteliConfiguration.pop_mailer_address,
+                 :port         => AlaveteliConfiguration.pop_mailer_port,
+                 :user_name    => AlaveteliConfiguration.pop_mailer_user_name,
+                 :password     => AlaveteliConfiguration.pop_mailer_password,
+                 :enable_ssl   => AlaveteliConfiguration.pop_mailer_enable_ssl }
+    self.settings = defaults.merge(values)
+    self.failed_mails = {}
+  end
+
+  def poll_for_incoming
+    found_mail = false
+    start do |pop3|
+      pop3.each_mail do |popmail|
+        found_mail = true
+        get_mail(popmail)
+      end
+    end
+    found_mail
+  end
+
+  def self.poll_for_incoming_loop
+    # Run poll_for_incoming in an endless loop, sleeping when there is
+    # nothing to do
+    while true
+      sleep_seconds = 1
+      while !poll_for_incoming
+        sleep sleep_seconds
+        sleep_seconds *= 2
+        sleep_seconds = 300 if sleep_seconds > 300
+      end
+    end
+  end
+
+  private
+
+  def get_mail(popmail)
+    unique_id = nil
+    raw_email = nil
+    received = false
+    begin
+      unique_id = popmail.unique_id
+      if retrieve?(unique_id)
+        raw_email = popmail.pop
+        RequestMailer.receive(raw_email)
+        received = true
+        popmail.delete
+      end
+    rescue Net::POPError, StandardError => error
+      if send_exception_notifications?
+        ExceptionNotifier.notify_exception(error, :data => { :mail => raw_email })
```

Comment:
Does `raw_email` contain the `unique_id`? If not, we should also send it so that we can easily match exception notifications to the `IncomingMessageError` records.